data/virt_autotest/setup_dns_service.sh

```diff
@@ -120,18 +120,26 @@ if [[ ${vmguest_failed} -ne 0 ]];then
 	exit 1
 fi
 #Wait for vm guests get assigned ip addresses
-sleep 90s
+sleep 60s
 echo -e "Virtual machines ${vm_guestnames_array[@]} have already been refreshed\n" | tee -a ${setup_log_file}
 
 #Write vm_hash_forward_ipaddr and vm_hash_reverse_ipaddr arrays
 for vmguest in ${vm_guestnames_array[@]};do
         get_vm_macaddress=`virsh domiflist --domain ${vmguest} | grep -oE "([0-9|a-z]{2}:){5}[0-9|a-z]{2}"`
         vm_macaddresses_array[${vm_hash_index}]=$(echo -e ${get_vm_macaddress})
+        for i in {1..30} ; do
+            get_vm_ipaddress=`tac $dhcpd_lease_file | awk '!($0 in S) {print; S[$0]}' | tac | grep -iE "${vm_macaddresses_array[${vm_hash_index}]}" -B8 | grep -oE "([0-9]{1,3}\.){3}[0-9]{1,3}" | tail -1`
+            [[ -n "$vm_ipaddress" ]] && break
+            sleep 10
+        done
```

Comment:
I would say that if guest gets ip address from internal dns service, it will not take such long to get a ip address. According to my observation, such guest can be fully up and running within 60 seconds.

Reply:
2 tests always failed since sle15sp7: 
https://10.145.10.207/tests/15612775#next_previous
https://10.145.10.207/tests/15612782#next_previous
They spent more time in getting an IP with unknown reasons.

Reply:
> How long do they need ?

I reproduced before the long holiday, it took nearly 3 minutes if I recalled correctly... I am not quite certain about the exact time since it has been too long since then.

> Is it this specific to SRIOV scenario (SRIOV attached network interface) ?

No, this module is ahead of SRIOV tests. It runs after downloading and restoring a guest disk.

Reply:
If this issue happens frequently,  is really significant and does not happen to other machines, I think you can do comparison with/out SRIOV settings ???  Or it only happens to certain machines regardless of SRIOV ???

Reply:
Only these sriov tests run both download guests and setup_dns. I guess the the enviroment related to downloading guests might be the reason.
An automation fix is always a better solution than grouping machines or tests.

Reply:
I think you'd better improve guests downloading if this is the case. Or say current guest downloading and restoring does not conform to the recommended steps. I think there should be something you can refer to.

Reply:
It is setup_dns that set up the environment including dhcp server and other stuff on the host, while downloading guests only copies the guest disk file from remote location and does nothing else. If there would be somenthing can be improved, it should be setup_dns rather than download_guests.

Reply:
What I said is the way in which downloaded guest is restored may not conform to recommended checkings/steps. Improvements can be done anywhere.

Reply:
This fix in the PR is the easiest way I have found so far. I am not against better ways you will find and figure it out why it took longer to get an IP than sle15sp6.

Reply:
I think you should have a deeper look even with this pull request. Because you do network SR-IOV on network SR-IOV host, and you found an issue relating to network.

Reply:
This issue blocked consequent sriov tests. With this PR, I had 2 successful sriov tests. I have not had more tests to check if there are more network problems. I will keep eyes if there are problems in future sriov tests.

Reply:
You can intervene manually a test run and check when there is resource.

Reply:
Sorry for jump in, if I understand correct, the test fails in setup_dns_service, and it's in higher priority for 15SP7,  @waynechen55 if you think this is not a good solution, since you are the author of this module, please take over the fixing and sooner is better to unblock the rest tests.

Reply:
What I mean is @Julie-CAO should be more careful about this possible and potential issue and take a deeper look at investigating the root cause. What I mean is a potential product issue needs to be attended relating to "SRIOV TEST RUN".

Reply:
> What I mean is a potential product issue needs to be attended relating to "SRIOV TEST RUN".

setup_dns run ahead of sriov test, this failure or change is nothing to to with sriov test. I made this change is because it blocked sriov tests.

Reply:
Thank you Wayne for the review!

I agree that a closer look may help locate root cause. Let me also add some background which you may not know. Before the national holiday, we were super busy, everyone on SP7 worked overtime -- lots of red tests/issues were there and there was long list TODO. Julie and Leon had to not only look into their problem, but also backup your areas. So indeed, to quickly shorten the TODO list, maybe there was not enough time spent on each issue, and some problems maybe hidden or may repeat in future runs. Glad that now you are back. Hope we can go back to a more normal-busy state, and everyone can have enough time to treat daily work in a more reasonable pace.

Reply:
I think it worth having a look because test run on a network SR-IOV enabled host. You can still have a deeper look with this pull request in place when there is resource.

Reply:
BTW @waynechen55 , how do you suggest to go for this PR now? Merge it and you debug separately to have sooner milestone result, or not merge it and wait for a potential new solution by you?

Reply:
> Thank you Wayne for the review!
> 
> I agree that a closer look may help locate root cause. Let me also add some background which you may not know. Before the national holiday, we were super busy, everyone on SP7 worked overtime -- lots of red tests/issues were there and there was long list TODO. Julie and Leon had to not only look into their problem, but also backup your areas. So indeed, to quickly shorten the TODO list, maybe there was not enough time spent on each issue, and some problems maybe hidden or may repeat in future runs. Glad that now you are back. Hope we can go back to a more normal-busy state, and everyone can have enough time to treat daily work in a more reasonable pace.

I think I gave my opinions clearly, but I do not mind doing it again.

I think it worth having a look because test run on a network SR-IOV enabled host. You can still have a deeper look with this pull request in place when there is resource.

Reply:
> BTW @waynechen55 , how do you suggest to go for this PR now? Merge it and you debug separately to have sooner milestone result, or not merge it and wait for a potential new solution by you?

I think I gave my opinions clearly, but I do not mind doing it again.

I think it worth having a look because test run on a network SR-IOV enabled host. You can still have a deeper look with this pull request in place when there is resource.

Reply:
setup_dns has been always running on SR-IOV enabled host for a long time ... If you think there is a bug or problem on sle15sp7 in setup_dns, it will be good that you can open a bug. but for sriov passthrough tests to guest, I did not find a bug so far.

Reply:
Yes. I think it worth having a look because test run on a network SR-IOV enabled host. You can still have a deeper look with this pull request in place when there is resource. I am happy if you confirm there is no issue at the moment or in the future.

Reply:
No, I will only take care of sriov test failure. About setup_dns problems, such as why the guest gets its IP longer than sle15sp6, it is beyond my knowledge and out of my fields.

Reply:
Yes. You can intervene test run manually and try boot vm manually on your SR-IOV enabled host to have a deeper look at this issue in the future.